src/components/TiltCard.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  useRef,
+  useState,
+  useCallback,
+  type CSSProperties,
+  type Ref,
+  type ReactNode,
+} from "react";
+import { classes } from "../utils/classes";
+
+interface ConditionalTiltCardProps extends TiltCardProps {
+  disabled?: boolean;
+}
+
+function ConditionalTiltCard({
+  children,
+  disabled = false,
+  maxTilt,
+  glareMaxOpacity,
+  className,
+  ref,
+  ...extras
+}: ConditionalTiltCardProps) {
+  if (disabled) {
+    return (
+      <div ref={ref} className={className} {...extras}>
+        {children}
+      </div>
+    );
+  }
+
+  return (
+    <TiltCard
+      ref={ref}
+      maxTilt={maxTilt}
+      glareMaxOpacity={glareMaxOpacity}
+      className={className}
+      {...extras}
+    >
+      {children}
+    </TiltCard>
+  );
+}
+
+interface TiltCardProps {
+  children: ReactNode;
+  maxTilt?: number;
+  glareMaxOpacity?: number;
+  className?: string;
+  ref: Ref<HTMLDivElement>;
+}
+
+// Light reflection angle based on tilt
+// Simulate light source coming from top-left (more natural)
+const lightSource = { x: -0.3, y: -0.5 };
+
+function TiltCard({
+  children,
+  maxTilt = 12,
+  glareMaxOpacity = 0.4,
+  className,
+  ref,
+  ...extras
+}: TiltCardProps) {
+  const cardRef = useRef<HTMLDivElement>(null);
+
+  const callbackRef = useCallback(
+    (node: HTMLDivElement | null) => {
+      if (ref) {
+        if (typeof ref === "function") ref(node);
+        else ref.current = node;
+      }
+      cardRef.current = node;
+    },
+    [ref],
+  );
+
+  const [tiltStyle, setTiltStyle] = useState<CSSProperties>({});
+  const [glareStyle, setGlareStyle] = useState<CSSProperties>({});
+  const [isHovered, setIsHovered] = useState(false);
+  const [tiltTransition, setTiltTransition] = useState("");
+  let transitionTimeout: NodeJS.Timeout;
```

Comment:
This needs to be a `useRef`, otherwise the value will not persist across re-renders.